### ISSUE TITLE
App may crash when load gif

### ISSUE THREAD

**Issue by xjy2061** [facebook/fresco#1706]
I think following code snippet in `gif.cpp` may cause crash when `pSavedImage->ImageDesc.Height` is 0.
```c++
// Check size of image.
if (pSavedImage->ImageDesc.Width <= 0 &&
    pSavedImage->ImageDesc.Height <= 0 &&
    pSavedImage->ImageDesc.Width > (INT_MAX / pSavedImage->ImageDesc.Height)) {
  return GIF_ERROR;
}
```

**Comment by oprisnik** [comment]
Thanks for the report @xjy2061. Do you have an example GIF that crashes?

**Comment by xjy2061** [comment]
http://p1.music.126.net/35Y3orENEaclRSwHUFWr1Q==/18983068253562273.gif
@oprisnik

**Comment by oprisnik** [comment]
I've fixed the issue in bcbe1c57f2d95be6f3827bf5f5a5a38cea5fec12. Thanks!
